pybamm/parameters/parameter_values.py

```diff
@@ -638,28 +662,60 @@ def _process_symbol(self, symbol):
+                if len(function_name) == 2:  # CSV data
+                    # to create an Interpolant
+                    name, data = function_name
+                    function = pybamm.Interpolant(
+                        data[:, 0], data[:, 1], *new_children, name=name
                     )
-                )
-                self.parameter_events.append(
-                    pybamm.Event(
-                        "Interpolant {} upper bound".format(name),
-                        pybamm.min(max(data[:, 0]) - new_children[0]),
-                        pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                    # Define event to catch extrapolation. In these events the sign is
+                    # important: it should be positive inside of the range and negative
+                    # outside of it
+                    self.parameter_events.append(
+                        pybamm.Event(
+                            "Interpolant {} lower bound".format(name),
+                            pybamm.min(new_children[0] - min(data[:, 0])),
+                            pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                        )
                     )
-                )
+                    self.parameter_events.append(
+                        pybamm.Event(
+                            "Interpolant {} upper bound".format(name),
+                            pybamm.min(max(data[:, 0]) - new_children[0]),
+                            pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                        )
+                    )
+
+                elif len(function_name) == 3:  # ML data
+                    # to create an Interpolant
+                    name, data, _ = function_name
+
+                    function = pybamm.Interpolant(
+                        data[0], data[-1], new_children, name=name
+                    )
+                    # Define event to catch extrapolation. In these events the sign is
+                    # important: it should be positive inside of the range and negative
+                    # outside of it
+                    for data_index in range(len(data[0]) - 1):
+                        self.parameter_events.append(
+                            pybamm.Event(
+                                "Interpolant {} lower bound".format(name),
+                                pybamm.min(new_children[data_index] - min(data[0][data_index])),
+                                pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                            )
+                        )
+                        self.parameter_events.append(
+                            pybamm.Event(
+                                "Interpolant {} upper bound".format(name),
+                                pybamm.min(max(data[0][data_index]) - new_children[data_index]),
+                                pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                            )
+                        )
```

Comment:
these lines could be written in such a way as to add these events for both 1D and 2D data, instead of repeating as they are now